common/djangoapps/student/tests/test_events.py

```diff
@@ -0,0 +1,112 @@
+
+
+class UserSettingsEventTestMixin(EventTestMixin):
+    """
+    Mixin for verifying that user setting events were emitted during a test.
+    """
+    def setUp(self):
+        super(UserSettingsEventTestMixin, self).setUp('util.model_utils.tracker')
+
+    def assert_user_setting_event_emitted(self, **kwargs):
+        """
+        Helper method to assert that we emit the expected user settings events.
+
+        Expected settings are passed in via `kwargs`.
+        """
+        self.assert_event_emitted(
+            USER_SETTINGS_CHANGED_EVENT_NAME,
+            table=self.table,  # pylint: disable=no-member
+            user_id=self.user.id,
+            **kwargs
+        )
+
+
+class TestUserProfileEvents(UserSettingsEventTestMixin, TestCase):
+    """
+    Test that we emit field change events when UserProfile models are changed.
+    """
+    def setUp(self):
+        super(TestUserProfileEvents, self).setUp()
+        self.table = 'auth_userprofile'
+        self.user = UserFactory.create()
+        self.profile = self.user.profile
+        self.reset_tracker()
+
+    def test_change_one_field(self):
+        """
+        Verify that we emit an event when a single field changes on the user
+        profile.
+        """
+        self.profile.year_of_birth = 1900
+        self.profile.save()
+        self.assert_user_setting_event_emitted(setting='year_of_birth', old=None, new=self.profile.year_of_birth)
+
+    def test_change_many_fields(self):
+        """
+        Verify that we emit one event per field when many fields change on the
+        user profile in one transaction.
+        """
+        self.profile.gender = u'o'
+        self.profile.bio = 'test bio'
+        self.profile.save()
+        self.assert_user_setting_event_emitted(setting='bio', old=None, new=self.profile.bio)
+        self.assert_user_setting_event_emitted(setting='gender', old=u'm', new=u'o')
+
+    def test_unicode(self):
+        """
+        Verify that the events we emit can handle unicode characters.
+        """
+        old_name = self.profile.name
+        self.profile.name = u'Dånîél'
+        self.profile.save()
+        self.assert_user_setting_event_emitted(setting='name', old=old_name, new=self.profile.name)
+
+
+class TestUserEvents(UserSettingsEventTestMixin, TestCase):
+    """
+    Test that we emit field change events when User models are changed.
+    """
+    def setUp(self):
+        super(TestUserEvents, self).setUp()
+        self.user = UserFactory.create()
+        self.reset_tracker()
+        self.table = 'auth_user'
+
+    def test_change_one_field(self):
+        """
+        Verify that we emit an event when a single field changes on the user.
+        """
+        old_username = self.user.username
+        self.user.username = u'new username'
+        self.user.save()
+        self.assert_user_setting_event_emitted(setting='username', old=old_username, new=self.user.username)
+
+    def test_change_many_fields(self):
```

Comment:
I'm thinking that the testing of multiple events vs. single events lives nearer to the helper methods.